delivery_gls_asm/models/delivery_carrier.py

```diff
@@ -0,0 +1,214 @@
+# Copyright 2020 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+from .gls_asm_request import GlsAsmRequest
+from .gls_asm_request import (
+    GLS_ASM_SERVICES, GLS_SHIPPING_TIMES, GLS_POSTAGE_TYPE)
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = "delivery.carrier"
+
+    delivery_type = fields.Selection(selection_add=[("gls_asm", "GLS ASM")])
+    gls_asm_uid = fields.Char(
+        string="GLS UID",
+    )
+    gls_asm_service = fields.Selection(
+        selection=GLS_ASM_SERVICES,
+        string="GLS Service",
+        help="Set the contracted GLS Service",
+        default="1",  # Courier
+    )
+    gls_asm_shiptime = fields.Selection(
+        selection=GLS_SHIPPING_TIMES,
+        string="Shipping Time",
+        help="Set the desired GLS shipping time for this carrier",
+        default="0",  # 10h
+    )
+    gls_asm_postage_type = fields.Selection(
+        selection=GLS_POSTAGE_TYPE,
+        string="Postage Type",
+        help="Postage type, usually 'Prepaid'",
+        default="P",
+    )
+
+    def _gls_asm_uid(self):
+        """The carrier can be put in test mode. The tests user must be set.
+           A default given by GLS is put in the config parameter data """
+        self.ensure_one()
+        uid = (
+            self.gls_asm_uid if self.prod_environment else
+            self.env['ir.config_parameter'].sudo().get_param(
+                'delivery_gls_asm.api_user_demo', ''))
+        return uid
+
+    def gls_asm_get_tracking_link(self, picking):
+        """Provide tracking link for the customer"""
+        tracking_url = ("http://www.asmred.com/extranet/public/"
```

Comment:
Can you store this url in carrier??

Reply:
Take a look at the implementation guidelines:

https://github.com/odoo/odoo/blob/12.0/addons/delivery/models/delivery_carrier.py#L27
https://github.com/odoo/odoo/blob/12.0/addons/delivery/models/delivery_carrier.py#L146-L150